v2/validator.go

```diff
@@ -73,3 +79,29 @@ func ValidateAutoEventFrequency(fl validator.FieldLevel) bool {
 	_, err := time.ParseDuration(fl.Field().String())
 	return err == nil
 }
+
+// ValidateUpdateDtoId used to check the UpdateDTO Id pointer value
+func ValidateUpdateDtoId(fl validator.FieldLevel) bool {
+	val := fl.Field()
+	// Since the Id should be string type, so skip the validation when it isn't a kind of string
+	if val.Kind() != reflect.String {
+		return true
+	}
+	_, err := uuid.Parse(val.String())
+	return err == nil
+}
+
+// ValidateUpdateDtoName used to check the UpdateDTO name pointer value
+func ValidateUpdateDtoName(fl validator.FieldLevel) bool {
+	val := fl.Field()
+	// Since the Name should be string type, so skip the validation when it isn't a kind of string
+	if val.Kind() != reflect.String {
+		return true
+	}
+	// The string value should not be empty
+	if len(strings.TrimSpace(val.String())) > 0 {
+		return true
+	} else {
+		return false
+	}
+}
```

Comment:
Since this are `*string` types, should be checking `val` for `nil` first?

And ` val.Kind() != reflect.String` will always be true since the type is `*string`.

Reply:
> Since this are *string types, should be checking val for nil first?

Yes, I modified the code to check the nil pointer.
```
if val.Kind() == reflect.Ptr && val.IsNil() 
```

> And val.Kind() != reflect.String will always be true since the type is *string

IF the pointer value is nil, then val.Kind() => ptr
IF the pointer value is string, then val.kind() => string